x-pack/plugins/observability/public/pages/slos/components/common/quick_filters.tsx

```diff
@@ -0,0 +1,127 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import React, { useEffect, useState } from 'react';
+import { AwaitingControlGroupAPI, ControlGroupRenderer } from '@kbn/controls-plugin/public';
+import { ViewMode } from '@kbn/embeddable-plugin/common';
+import styled from 'styled-components';
+import { Filter } from '@kbn/es-query';
+import { useCreateDataView } from '../../../../hooks/use_create_data_view';
+import { SearchState } from '../../hooks/use_url_search_state';
+import { SLO_SUMMARY_DESTINATION_INDEX_NAME } from '../../../../../common/slo/constants';
+
+interface Props {
+  initialState: SearchState;
+  loading: boolean;
+  onStateChange: (newState: Partial<SearchState>) => void;
+}
+
+export function QuickFilters({ initialState: { tagsFilter, statusFilter }, onStateChange }: Props) {
+  const { dataView, loading } = useCreateDataView({
+    indexPatternString: SLO_SUMMARY_DESTINATION_INDEX_NAME,
+  });
+  const [controlGroupAPI, setControlGroupAPI] = useState<AwaitingControlGroupAPI>();
+
+  useEffect(() => {
+    if (!controlGroupAPI) {
+      return;
+    }
+    const subscription = controlGroupAPI.onFiltersPublished$.subscribe((newFilters) => {
+      if (newFilters.length === 0) {
+        onStateChange({ tagsFilter: undefined, statusFilter: undefined });
+      } else {
+        onStateChange({
+          tagsFilter: newFilters.filter((filter) => filter.meta.key === 'slo.tags')?.[0],
+          statusFilter: newFilters.filter((filter) => filter.meta.key === 'status')?.[0],
+        });
+      }
+    });
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [controlGroupAPI, onStateChange]);
+
+  if (loading || !dataView) {
+    return null;
+  }
+
+  return (
+    <Container>
+      <ControlGroupRenderer
+        getCreationOptions={async (initialInput, builder) => {
+          await builder.addOptionsListControl(initialInput, {
+            dataViewId: dataView.id!,
+            fieldName: 'status',
+            width: 'small',
+            grow: true,
+            title: STATUS_LABEL,
+            controlId: 'slo-status-filter',
+            exclude: statusFilter?.meta.negate,
+            selectedOptions: getSelectedOptions(statusFilter),
+            existsSelected: Boolean(statusFilter?.query?.exists.field === 'status'),
+            placeholder: ALL_LABEL,
+          });
+          await builder.addOptionsListControl(initialInput, {
+            dataViewId: dataView.id!,
+            title: TAGS_LABEL,
+            fieldName: 'slo.tags',
+            width: 'small',
+            grow: false,
+            controlId: 'slo-tags-filter',
+            selectedOptions: getSelectedOptions(tagsFilter),
+            exclude: statusFilter?.meta.negate,
+            existsSelected: Boolean(tagsFilter?.query?.exists.field === 'slo.tags'),
+            placeholder: ALL_LABEL,
+          });
```

Comment:
Looks like you can run these promises in parallel.

```suggestion
         await Promise.all(builder.addOptionsListControl(...), builder.addOptionsListControl(...))
```

Reply:
i think it's micro optimization in this context, so i am going to leave it as is.